include/sundials/sundials_math.h

```diff
@@ -75,7 +75,7 @@ extern "C" {
  */
 
 #ifndef SUNRsqrt
-#if defined(SUNDIALS_USE_GENERIC_MATH)
+#if (__STDC_VERSION__ < 199901L) && defined(SUNDIALS_USE_GENERIC_MATH)
```

Comment:
This might cause an issue with compilers that do not define `__STDC_VERSION__`. I think with the suggested edit to default `USE_GENERIC_MATH` to `OFF` when `CMAKE_C_STANDARD` is not `90` the checks on `__STDC_VERSION__` can be removed.

Reply:
Ok, I'll revert adding this check.  It looks like a couple unrelated function use a similar check:
https://github.com/LLNL/sundials/blob/e2f29c34f324829302037a1492db480be8828084/src/sundials/sundials_math.c#L25-L32
Perhaps these should also be based on `USE_GENERIC_MATH`

Reply:
Not sure why these have not caused a problem. Perhaps @balos1 recalls what the specific issue was with `armclang` and relying on `__STDC_VERSION__` to define `sunindextype`.

Reply:
I dont recall exactly the `armclang` issue, but `__STDC_VERSION__` may not be defined by some C89/C90 compliant compilers or C++ compilers, however `__STDC__` would be defined for C89/C90 compliant compilers. So the checks should be something like:

```
#if !defined(__cplusplus) && defined(__STDC_VERSION__)
#if    __STDC_VERSION__ >= 199901L
// C99
#endif
#elif (__STDC__)
// C89/C90 for sure
#elif defined(__cplusplus)
// C++ -- what to do
#else
#warning non-compliant C compiler 
#endif
```

And there is the possibility of MSVC and partial support for some C standard newer than C89. Sigh. @Steven-Roberts thanks for noticing the use of ` #if (__STDC_VERSION__ >= 199901L)` in the other sundials_math functions. I think we need to probably change these checks so that CMake instead does a compile check to see if the function is available.  I can add that to this PR.

Finally, I think we should keep `USE_GENERIC_MATH=ON` by default for all standards and then just warn + document for the C90 case.